ui/src/app/pages/workspace/invalid-billing-banner.tsx

```diff
@@ -19,23 +21,119 @@ interface Props extends NavigationProps {
   onClose: Function;
 }
 
+const InitialCreditsArticleLink = () => (
+  <>
+    "
+    <LinkButton
+      onClick={() => window.open(supportUrls.createBillingAccount, '_blank')}
+    >
+      Using <AoU /> Initial Credits
+    </LinkButton>
+    "
+  </>
+);
+
+const BillingAccountArticleLink = () => (
+  <>
+    "
+    <LinkButton
+      onClick={() => window.open(supportUrls.createBillingAccount, '_blank')}
+    >
+      Paying for Your Research
+    </LinkButton>
+    "
+  </>
+);
+
 export const InvalidBillingBanner = fp.flow(
   withCurrentWorkspace(),
   withUserProfile(),
   withNavigation
-)(({ onClose, navigate, workspace }: Props) => {
-  const message = (
-    <div>
-      The initial credits for the creator of this workspace have run out. Please
-      provide a valid billing account.
-      <LinkButton
-        onClick={() => window.open(supportUrls.createBillingAccount, '_blank')}
-      >
-        Learn how to link a billing account.
-      </LinkButton>
-    </div>
-  );
-  const footer = (
+)(({ onClose, navigate, workspace, profileState }: Props) => {
+  const { profile } = profileState;
+  const isCreator =
+    workspace && profile && profile.username === workspace.creator;
+  const isEligibleForExtension = workspace && !workspace.extensionEpochMillis;
+  const isExpired =
+    workspace && workspace.initialCredits.expirationEpochMillis < Date.now();
+  const isExpiringSoon =
+    workspace &&
+    !isExpired &&
+    plusDays(workspace.initialCredits.expirationEpochMillis, 5) < Date.now();
+  let message;
+  let title;
+  if (isExpiringSoon && isEligibleForExtension) {
```

Comment:
This feels complicated, but the only way that I can think of simplifying it is to break up individual phrases into variables and then have several ternary statements. I feel like that might be more efficient, but I'm not sure it would be more readable.

I would love thoughts here.

Reply:
this looks outdated.  is there a similar comment for the updated version?